developer-guides/miscellaneous-guides/multi-message-transaction.md

```diff
@@ -0,0 +1,237 @@
+
+A transaction, on the other hand, is a collection of one or more messages that are grouped together and signed by a single account. When a transaction is broadcast to the network, it is validated by the nodes on the blockchain, and if it is deemed valid, the requested actions specified in the messages are executed. Transactions provide a way to ensure that multiple actions can be performed atomically, meaning that either all of the actions are executed successfully or none of them are.
+
+One transaction per block, with a transaction being multiple messages.
+
+## Transaction Limitation / Account Sequence
+
+By default, you can only submit 1 transaction per block. In doing this, it will increase your wallets account sequence by one. If you try to submit multiple transactions in the same block, it will error with a account sequence mismatch error. This means there is a pending transaction for your account already in the on chain pending pool (mempool of transactions). Once this transaction goes through and all messages are successfully executed, you can submit another transaction in the following block.
+
+## Multi Message Transaction (bank)
+
+To generate a multiple message transaction, you will need to slight scripting knoladge to assist for your needs. This works for any module in Juno, and you can add as many different messages to the same transaction within the gas limit. (Gas is a deterministic number describing the amount of computer a given action takes on a node).
+
+First, you need to generate a default single message transaction from the command line like so
+
+```sh
+junod tx bank send juno16g2r... juno1t8e... 1000000ujuno --generate-only
+```
+
+```json
+// this generates the following. You can pipe the output into 
+// the `jq` command on mac / linux to format it like this
+{
+  "body": {
+    "messages": [
+      {
+        "@type": "/cosmos.bank.v1beta1.MsgSend",
+        "from_address": "juno16g2rahf5846rxzp3fwlswy08fz8ccuwk03k57y",
+        "to_address": "juno1t8ehvswxjfn3ejzkjtntcyrqwvmvuknzy3ajxy",
+        "amount": [
+          {
+            "denom": "ujuno",
+            "amount": "1000000"
+          }
+        ]
+      }
+    ],
+    "memo": "",
+    "timeout_height": "0",
+    "extension_options": [],
+    "non_critical_extension_options": []
+  },
+  "auth_info": {
+    "signer_infos": [],
+    "fee": {
+      "amount": [],
+      "gas_limit": "200000",
+      "payer": "",
+      "granter": ""
+    }
+  },
+  "signatures": []
+}
+```
+
+Here are are interested in the body -> messages section. In this array, we can put any amount of actions (within the gas limitation) for this transaction to execute with a single execute. So if I wanted to send another amount to a different address at the same time as well, we can manually alter the array of messages to add this new message like so
+
```

Comment:
Maybe show to manually add to the array for example something like 

```Create a file txs.json containing the JSON array:``` 

i created this on uni-6 

```{
  "body": {
    "messages": [
      {
        "@type": "/cosmos.bank.v1beta1.MsgSend",
        "from_address": "juno1p4ar7sytjgjhhcvfnf298ytgklxrgnx4m0zwm2",
        "to_address": "juno1xwazl8ftks4gn00y5x3c47auquc62ssune9ppv",
        "amount": [
          {
            "denom": "ujunox",
            "amount": "1000000"
          }
        ]
      },
      {
        "@type": "/cosmos.bank.v1beta1.MsgSend",
        "from_address": "juno1p4ar7sytjgjhhcvfnf298ytgklxrgnx4m0zwm2",
        "to_address": "juno15twk6xu5rnrrlnf7c5zy92gvykcs4h5ucxzzqq",
        "amount": [
          {
            "denom": "ujunox",
            "amount": "5000000"
          }
        ]
      }
    ],
    "memo": "",
    "timeout_height": "0",
    ...
  }``` 

named it as ```multi_tx_json``` 

i then tried to run ```junod tx broadcast multi_tx.json --chain-id uni-6 --from Highlander --yes --fees 5000ujunox | jq``` 

and get a parse error ```Error: invalid character '.' looking for beginning of object key string: tx parse error
Usage:``` 

Would love your opinion on if im doing this the right way or if im missing something else here. :)

Reply:
Ahh yes I cut out the bottom part so the ...'s there should not be used. Will give a full copy here